Add JSON user API (#17352)
pypi/warehouse: docs/user/api/json.md

}
```

### Get a user

Route: `GET /user/<username>/json`

Returns the same information found in the HTML profile page (`/user/<username>`), but in a JSON format. 

> It contains the time of account creation, the username, the name (or null), and a list of the user's projects.

Reviewer: ```suggestion
It contains the time of account creation, the username, the name (if present), and a list of the user's projects.
```

Status codes:

* `200 OK` - no error
* `404 Not Found` - User was not found

Example request:

> ```
> GET /user/someuser/json HTTP/1.1
> Host: pypi.org
> Accept: application/json
> ```

Reviewer: Nitpick: spacing + fence syntax:

```suggestion

```http
GET /user/someuser/json HTTP/1.1
Host: pypi.org
Accept: application/json
```
```

> 

Reviewer 2: Missing end of code fence?

??? "Example JSON response"

>     ```http

Reviewer 2: Wrong indent? Should be 0.
